core/tests/test_sql_performance.py

```diff
@@ -32,3 +32,42 @@ def test_sql_benchmark(batchsize:int,expected_ops_per_sec:int,fields:int):
     actual_ops_per_sec = cycles/(end-start)
     logging.debug(f"ops per sec: {actual_ops_per_sec}")
     assert actual_ops_per_sec>expected_ops_per_sec
+
+@pytest.mark.parametrize("batchsize,expected_ops_per_sec,fields", [(100,40000,20), (1000, 40000,20), (10000, 40000,20)])
+def test_sql_benchmark_nested(batchsize:int,expected_ops_per_sec:int,fields:int):
+    """ A rough sanity benchmark to test the ballpark figures of the expression language
+
+    Args:
+        batchsize (int): size of batch
+        expected_ops_per_sec (int): expected operations per sec
+    """
+    # suppress logging
+    logging.getLogger("dy").disabled = True
+    expression_text = "upper(`a.b.c.d`)"
+    expr = expression.compile(expression.Language.SQL.value,expression_text)
+    cycles = 200000
+    start = time.time()
+    batch = 1000
+
+    # dummy record with 20 fields
```

Comment:
seems it has more than 20 fields in total (all nested levels)

Reply:
changed